README.md

```diff
@@ -36,3 +36,157 @@ This is the shared library defining public classes that constitute the worker in
+```
+function onProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+This function calls the customization scripts, and if none of them have set the cancellation flag then calls the implementation's `DocumentWorker#processDocument processDocument()` function. 
+
+#### onAfterProcessDocument
+
+This function will be called once the processing of the document completed successfully.
+This function is passed `DocumentEventObject` with `Document` as an argument.
+
+An example of the `DocumentEventObject` is explained in `onProcessDocument` section.
+
+For more details  of the event object,  refer the [onProcessDocument](#onProcessDocument) section. 
+
+```
+function onAfterProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+#### onAfterProcessTask
+
+This is the last function called by worker on the task message.
+This function is passed `TaskEventObject` with `Task` as an argument.
+
+An example of `TaskEventObject` explained in `onProcessTask` section. 
+
+The variables will be initialized with values from task message and it will be sent to the series of other functions in the workflow for further processing.
+
+For more details of the `TaskEventObject`, refer the [onProcessTask](#onProcessTask) section.
+
+```
+function onAfterProcessTask(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // onProcessDocument(e.rootDocument)
+}
+```
+#### onError
+
+This function should be called in case of a failure in the worker that is not handled by the worker code. In chained workers, this will allow continuing to process the document.
+This function is passed `ErrorEventObject` with `Task`and `RuntimeException` as an argument.
+
+```
+ErrorEventObject(Task task, RuntimeException error){
+    super(task);
+    error = error;
+    handled = false;
+}
+```
+For more details of the `ErrorEventObject` refer the java implementation of the class for the class [ErrorEventObject](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/ErrorEventObject.java)
+
+```
+function onError(errorEvent)
+{
+     // errorEvent.application  (read-only)
+     // errorEvent.task         (read-only)
+     // errorEvent.rootDocument (read-only)
+     // errorEvent.handled      (writable)  (default: false)
+}
+```
+Output of the function would be the boolean flag whether the failure was handled.
+
+If it is not handled by event handler, exception will be thrown from worker.
```

Comment:
What is the outcome of this exception being thrown? Is the message placed on a special queue in rabbit? What is that messages status?

Reply:
Updated about message status after failure.